src/libs/ReportActionsUtils.ts

```diff
@@ -45,15 +46,6 @@ type MemberChangeMessageRoomReferenceElement = {
 
 type MemberChangeMessageElement = MessageTextElement | MemberChangeMessageUserMentionElement | MemberChangeMessageRoomReferenceElement;
 
-let allReports: OnyxCollection<Report> = {};
-Onyx.connect({
```

Comment:
Can you explain this in more detail?

Reply:
@parasharrajat It's true that Onyx caches value for the keys but each time we are adding a connection the values will be accessed from the cache if present otherwise from the storage. Issue arises when we are dealing with huge collections like Report. On heavy accounts we have ~15k reports, so for each connection accessing the keys from cache/ storage and then sending them to the client adds up to the app startup. I am attaching a screenshot of a profile before the changes in this PR, where we can clearly see how `fastMerge` is called recursively for a longer period of time. This is also explained in the PR description.

Before:

![before](https://github.com/Expensify/App/assets/47336142/f2a20932-3281-4d17-8c8f-e4f3ab27634c)

After:

![after](https://github.com/Expensify/App/assets/47336142/33075faf-55a4-4ad5-a228-9ec052b81fff)

In after, we are keeping only one connection to Report in `ReportConnection`.